machine/machine.go

```diff
@@ -313,3 +313,57 @@ func getNumCores() int {
 
 	return numCPU
 }
+
+var cpuToSocket map[int]int
+
+// GetSocketFromCPU returns socket which is bounded to passed CPU.
+// If there is no information it would return -1.
+func GetSocketFromCPU(cpu int) int {
+	// Check if information is cached.
+	if cpuToSocket == nil {
+		var err error
+		cpuToSocket, err = readCPUToSocket(cpuBusPath)
+		if err != nil {
+			klog.Errorf("Couldn't gather information about CPU bounded to socket: %v", err)
+		}
+	}
+
+	if _, ok := cpuToSocket[cpu]; !ok {
+		return -1
+	}
+
+	return cpuToSocket[cpu]
+}
+
+func readCPUToSocket(basePath string) (map[int]int, error) {
```

Comment:
This function also retrives infromation from sysfs so it fits better to [sysinfo](https://github.com/google/cadvisor/tree/master/utils/sysinfo) or [sys](https://github.com/google/cadvisor/tree/master/utils/sysfs), please move it if it is possible.

Reply:
Functionality moved to ``utils/sysinfo/GetSocketFromCPU``.